### PR TITLE
Asst ds 5

### PR DESCRIPTION
**Pull request by russellschmidt**

### CODE REVIEW

File: 01-data-structures/05-hashes-part-2/open_addressing/open_addressing.rb [hunk 1]
```diff
@@ -2,29 +2,85 @@
...
+      end
+    else
+      resize
+      add_element(key, value)
+    end
   end
 
   def [](key)
+    items_index = index(key, self.size)
+    next_index = @items[items_index].next
```
Comment (dawsonc623) [inline 01-data-structures/05-hashes-part-2/open_addressing/open_addressing.rb:39]: The `next` construct is not actually part of the open addressing algorithm. It should have never been in the open addressing `Node` class. However, your usage of it is clever, as it avoids the looping. Keep in mind, however, that if that "next" was ever deleted, you would have to find if any node was pointing at it and set those `next` properties to `nil` as well.

Comment (russellschmidt) [reply 01-data-structures/05-hashes-part-2/open_addressing/open_addressing.rb:39]: I see. So instead of using `next` I could have done `(x...arr_length).each {|x| (x.nil?)? x = y}` where x is the index to go hunt for open slots. Of course I would not use that overly terse, hard to follow syntax presented here only for discussion's sake. Your way makes more sense. My implementation is a little more convoluted in that yes I take up more memory on every element and also have to write and maintain code to support the `next` instance variable.

Comment (dawsonc623) [reply 01-data-structures/05-hashes-part-2/open_addressing/open_addressing.rb:39]: As we experienced today, adding in those extra pointers complicates implementations a bit more than it would seem on the surface. In the end, you end up just trading one complexity for another, usually.

**Comment by dawsonc623** [comment]
This branch is merged and can be deleted.
